### PR TITLE
interfaces/builtin: add autopilot-introspection

### PR DESCRIPTION
**Pull request by heber013**
Adding AutopilotIntrospectionInterface that adds the rules for enabling the use of dbus for introspecting a snap application using Autopilot, which will allow to run automated tests against snaps.

### CODE REVIEW

File: interfaces/builtin/autopilot_introspection.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+ */
+
+package builtin
+
+import (
+	"github.com/snapcore/snapd/interfaces"
+)
+
+const introspectionConnectedPlugAppArmor = `
+# Description: Can introspect an application with a testability library.
```
Comment (zyga) [inline interfaces/builtin/autopilot_introspection.go]: Is this about being introspected or actually introspecting?

File: interfaces/builtin/autopilot_introspection.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    interface=com.canonical.Autopilot.Introspection
+    member=GetVersion
+    peer=(label=unconfined),
+
+dbus (send, receive)
+    bus=session
+    path=/com/canonical/Autopilot/Introspection
+    interface=com.canonical.Autopilot.Introspection
+    member=GetState
+    peer=(label=unconfined),
```
Comment (zyga) [inline interfaces/builtin/autopilot_introspection.go]: In each those cases the peer is unconfined so it must come from classic. If that's the design you want then I'd suggest adjusting the documentation and documenting this in `docs/interfaces.md`

**Comment by niemeyer** [comment]
Why do we need an interface for this, rather than having autopilot controlling the whole machine for testing the snaps?

**Comment by heber013** [comment]
We were reviewing this approach, as Gustavo said, it is not necessary to expose a new interface to update the appamor rules for enabling Autopilot introspection. Instead, we will take another alternative that is creating an API in autopilot that enables the appamor profiles of the applications that need to be tested.

**Comment by heber013** [comment]
@niemeyer @zyga :
Hey guys, we were reviewing the approach to get the appamor rules updated to allow the snaps applications to be introspected by autopilot. We think that the approach of delegating this to autopilot itself would work in classic installations. But also we are thinking in a fully confined environment, in this case: Will it be possible to update the apparmor rules through autopilot? Or will we need an interface for achieving that? In this case the snap under test should have the plug defined, and in the other side autopilot (or any other testing tool) offers the slot to allow introspection. Let me know your thoughts and if our reasoning make sense.

**Comment by niemeyer** [comment]
Please don't fiddle with the apparmor rules behind snapd's back. This is an implementation detail from the system, which is changing all the time and has no promises of stability.

If you own the whole system and is capable of changing apparmor rules, it also means the process is running unconfined and can effectively do whatever it needs to do for testing.

I'm still missing the larger context in which this is taking place. I feel like trying to reverse engineering the problem out of the implementation.

**Pull request closed without merging**